lib/Mail/DMARC.pm

```diff
@@ -130,6 +133,15 @@ sub _from_mail_dkim {
                 human_result => $s->result_detail,
             );
     }
+
+    if ($signatures < 1) {
+        push @{ $self->{dkim}},
+            Mail::DMARC::Report::Aggregate::Record::Auth_Results::DKIM->new(
```

Comment:
I don't like adding a fake domain here, it feels wrong, and could end up in reports which would be confusing. It's better if we make the is_valid method of ::DKIM allow domain to be undef when result is none, and then make sure that any dependent code does the right thing with that.

Reply:
Good point about the fake domain. If there was to be a fake domain, it should definitely end with a reserved TLD such as `.invalid`. Perhaps using an empty string is better?  And then test for defined-ness?

Reply:
Yes, that should work better